lib/ansible/parsing/mod_args.py

```diff
@@ -304,4 +304,8 @@ def parse(self):
                                                                                                                                   ", ".join(RAW_PARAM_MODULES)),
                                          obj=self._task_ds)
 
+        # common user error
+        if any(arg.startswith('with_') for arg in args):
```

Comment:
modules already return an error  on unknown options, its better to capture there and then look against 'all keywords' not just `with_`

Reply:
Agreed, I don't think that we can do this as we don't really have reserved words like a programming language.

This seems like it belongs in a linter?

Reply:
@sivel as per your comment, this can be closed and dealt with when the work you mentioned is done?

Reply:
@mkrizek Likely, however I cannot guarantee when that work will land. Likely not until 2.7